tests/test_outputs_handler_matsim_xml_writer.py

```diff
@@ -459,3 +459,27 @@ def test_write_matsim_vehicles_produces_semantically_equal_xml_to_input_matsim_x
     matsim_xml_writer.write_vehicles(tmpdir, network.schedule.vehicles, network.schedule.vehicle_types)
 
     xml_diff.assert_semantically_equal(os.path.join(tmpdir, 'vehicles.xml'), pt2matsim_vehicles_file)
+
+
+def test_network_with_elevation_data_produces_valid_matsim_network_xml_file(tmpdir, network_dtd):
+    network = Network('epsg:27700')
+    network.add_node('0', attribs={'id': '0', 'x': 1, 'y': 2, 'z': 3, 'lat': 1, 'lon': 2})
+    network.add_node('1', attribs={'id': '1', 'x': 2, 'y': 2, 'z': 0, 'lat': 2, 'lon': 2})
+    network.add_link('0', '0', '1', attribs={'id': '0', 'from': '0', 'to': '1', 'length': 1, 'freespeed': 1,
+                                             'capacity': 20, 'permlanes': 1, 'oneway': '1', 'modes': ['car'],
+                                             'extra_Special_attrib': 12})
+    network.write_to_matsim(tmpdir)
+    generated_network_file_path = os.path.join(tmpdir, 'network.xml')
+    xml_obj = lxml.etree.parse(generated_network_file_path)
+    assert network_dtd.validate(xml_obj), \
+        'Doc generated at {} is not valid against DTD due to {}'.format(generated_network_file_path,
+                                                                        network_dtd.error_log.filter_from_errors())
+    _network_from_file = read.read_matsim(path_to_network=generated_network_file_path, epsg='epsg:27700')
+    assert_semantically_equal(dict(_network_from_file.nodes()), {
+        '0': {'id': '0', 'x': 1.0, 'y': 2.0, 'z': 3.0, 'lon': -7.557148039524952, 'lat': 49.766825803756994,
+              's2_id': 5205973754090365183},
+        '1': {'id': '1', 'x': 2.0, 'y': 2.0, 'z': 0.0, 'lon': -7.557134218911724, 'lat': 49.766826468710484,
+              's2_id': 5205973754090480551}})
+    assert_semantically_equal(dict(_network_from_file.links()), {
+        '0': {'id': '0', 'from': '0', 'to': '1', 'freespeed': 1.0, 'capacity': 20.0, 'permlanes': 1.0, 'oneway': '1',
+              'modes': {'car'}, 's2_from': 5205973754090365183, 's2_to': 5205973754090480551, 'length': 1.0}})
```

Comment:
split this portion into a separate test: `test_saving_network_with_elevation_data_to_xml_file_saves_correct_z_value_data`

I can see this test was inspired by this test: `test_network_with_extra_attribs_produces_valid_matsim_network_xml_file` which should also be split into 2 🤷   you can get rid of `'extra_Special_attrib': 12` that was what was tested in `test_network_with_extra_attribs_produces_valid_matsim_network_xml_file` and you don't have to test values of `_network_from_file.links()` because there is nothing new there.